componentspush.ps1

```diff
@@ -0,0 +1,51 @@
+﻿Param(
+  [string]$libraryId = "",
+  [string]$apiKey = "",
+  [string]$endpoint = "https://components-api.sitecorecloud.io",
+  [string]$rootFolder = ".\src\components"
+)
+
+$headers = @{
+    "x-api-key"=$apiKey
+}
+
+# Components
+$collections = Get-Content "$rootFolder\collections.json" | ConvertFrom-Json
+foreach ($collection in $collections)
+{
+    Write-Host "Collection: " $collection.Name
+
+   foreach ($component in $collection.components)
+   {
+      Write-Host "  Component: " $component.Name
+
+      $componentUrl = "$endpoint/libraries/$libraryId/collections/$($collection.id)/components"
+      Invoke-WebRequest -Uri $componentUrl -Method Post -Body ($component | ConvertTo-Json -depth 100) -Headers $headers -ContentType "application/json"
```

Comment:
@adoprog I am getting errors when pushing serialized components to a new XM Cloud organization. Specifically if the serialized content contains collections that do not exist on the new XM Cloud organization:
```
Collection:  Static Components
  Component:  Promo
Invoke-WebRequest : {"statusCode":404,"error":"Not Found","message":"Could not find collection"}
At C:\projects\Sitecore.Demo.XmCloud.PlaySummit\componentspush.ps1:23 char:7
+       Invoke-WebRequest -Uri $componentUrl -Method Post -Body ($compo ...
+       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-WebRequest], WebException
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeWebRequestCommand
    Version:  Name ( staged )
```

We should create the collections first if they do not already exist.